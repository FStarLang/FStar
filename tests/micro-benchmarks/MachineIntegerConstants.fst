module MachineIntegerConstants

(*
  Check that all machine integer constants from modules [U]Int(8-64)
  extract to Stdint in OCaml.
  The only exception is UInt8, that should extract to OCaml's native
  int type.
*)

// Signed

let e_int8 x = match x with
  | 22y -> FStar.Int8.(42y +^ x)
  | _ -> 0y

let e_int16 x = match x with
  | 22s -> FStar.Int16.(42s +^ x)
  | _ -> 0s

let e_int32 x = match x with
  | 22l -> FStar.Int32.(42l +^ x)
  | _ -> 0l

let e_int64 x = match x with
  | 22L -> FStar.Int64.(42L +^ x)
  | _ -> 0L

// Unsigned

let e_uint8 x = match x with
  | 22uy -> FStar.UInt8.(42uy +^ x)
  | _ -> 0uy

let e_uint16 x = match x with
  | 22us -> FStar.UInt16.(42us +^ x)
  | _ -> 0us

let e_uint32 x = match x with
  | 22ul -> FStar.UInt32.(42ul +^ x)
  | _ -> 0ul

let e_uint64 x = match x with
  | 22UL -> FStar.UInt64.(42UL +^ x)
  | _ -> 0UL
