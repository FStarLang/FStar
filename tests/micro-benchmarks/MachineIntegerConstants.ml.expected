open Prims
let (e_int8 : FStar_Int8.t -> FStar_Int8.t) =
  fun x ->
    match x with
    | uu___ when uu___ = (Stdint.Int8.of_int (22)) ->
        FStar_Int8.add (Stdint.Int8.of_int (42)) x
    | uu___ -> Stdint.Int8.zero
let (e_int16 : FStar_Int16.t -> FStar_Int16.t) =
  fun x ->
    match x with
    | uu___ when uu___ = (Stdint.Int16.of_int (22)) ->
        FStar_Int16.add (Stdint.Int16.of_int (42)) x
    | uu___ -> Stdint.Int16.zero
let (e_int32 : FStar_Int32.t -> FStar_Int32.t) =
  fun x ->
    match x with
    | uu___ when uu___ = (Stdint.Int32.of_int (22)) ->
        FStar_Int32.add (Stdint.Int32.of_int (42)) x
    | uu___ -> Stdint.Int32.zero
let (e_int64 : FStar_Int64.t -> FStar_Int64.t) =
  fun x ->
    match x with
    | uu___ when uu___ = (Stdint.Int64.of_int (22)) ->
        FStar_Int64.add (Stdint.Int64.of_int (42)) x
    | uu___ -> Stdint.Int64.zero
let (e_uint8 : FStar_UInt8.t -> FStar_UInt8.t) =
  fun x ->
    match x with | uu___ when uu___ = 22 -> FStar_UInt8.add 42 x | uu___ -> 0
let (e_uint16 : FStar_UInt16.t -> FStar_UInt16.t) =
  fun x ->
    match x with
    | uu___ when uu___ = (Stdint.Uint16.of_int (22)) ->
        FStar_UInt16.add (Stdint.Uint16.of_int (42)) x
    | uu___ -> Stdint.Uint16.zero
let (e_uint32 : FStar_UInt32.t -> FStar_UInt32.t) =
  fun x ->
    match x with
    | uu___ when uu___ = (Stdint.Uint32.of_int (22)) ->
        FStar_UInt32.add (Stdint.Uint32.of_int (42)) x
    | uu___ -> Stdint.Uint32.zero
let (e_uint64 : FStar_UInt64.t -> FStar_UInt64.t) =
  fun x ->
    match x with
    | uu___ when uu___ = (Stdint.Uint64.of_int (22)) ->
        FStar_UInt64.add (Stdint.Uint64.of_int (42)) x
    | uu___ -> Stdint.Uint64.zero